### PR TITLE
Splash screen allows exiting early (with escape key)

### PR DESCRIPTION
**Pull request by will-hart**
This PR allows the player to exit the splash screen at any time by hitting `KeyCode::escape`.

---

Previously this PR also contained code for displaying multiple splash screens, but after discussion that will be migrated to an example in the bevy repository.

### CODE REVIEW

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:60]: This can be a `Vec` instead of a `VecDeque`.

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:68]: Can `impl FromWorld` instead, to get access to `world.resource::<AssetServer>()`. Then loading the images can be done here.

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:56]:
```suggestion
#[derive(Component, Reflect)]
#[reflect(Resource)]
```

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:59]:
```suggestion
#[derive(Resource, Reflect)]
#[reflect(Resource)]
```

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:192]: `let-else` then set the next screen and `return` here to remove one level of indentation.

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:194]: `let-else` here as well.

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:1]: Comments should begin with a capital letter and end with a `.` for style consistency.

**Review by benfrankel — CHANGES_REQUESTED** [review]

File: src/screens/splash.rs
Comment (benfrankel) [reply src/screens/splash.rs:56]: Also make sure to add `app.register_type` to `plugin`.

File: src/screens/splash.rs
Comment (benfrankel) [reply src/screens/splash.rs:59]: Also make sure to add `app.register_type` to `plugin`.

File: src/screens/splash.rs
Comment (will-hart) [reply src/screens/splash.rs:60]: I may be missing something but I feel like `VecDeque` is the better option here as we're popping from the front. 

There is `Vec::pop` which pops from the back, meaning we'd show the images in reverse order. The docs for `pop` actually say:

> If you'd like to pop the first element, consider using VecDeque::pop_front instead.

or there is something like `Vec::drain` but that returns an iterator so there is a lot more boilerplate.

Comment (benfrankel) [reply src/screens/splash.rs:60]: That's true. We would either need to store the images in reverse order, use `vec.remove(0)` (when we know the vec is non-empty), or use an index instead of removing the images from the list. The index could reset to 0 when entering `Screen::Splash`.

I normally don't see `VecDeque` unless it's a performance optimization, which isn't relevant in this case.

File: src/screens/splash.rs
Comment (will-hart) [reply src/screens/splash.rs:194]: Thanks - I forgot there is `let-else` 😁 

The container *should* exist, so what do you think about just doing `get()` here?

File: src/screens/splash.rs
Comment (benfrankel) [reply src/screens/splash.rs:60]: I'll defer to other opinions on this.

File: src/screens/splash.rs
Comment (benfrankel) [reply src/screens/splash.rs:194]: I'm an `unwrap` hater so I'd say just do `let-else`, and in my own code I would do `let container = r!(containers.get_single());` using the shorthand `or_return!` macro from `tiny_bail`.

Comment (benfrankel) [reply src/screens/splash.rs]: Oh actually after a second look, I would rename `containers` -> `container_query` and make this a `for` loop. No need to assume / check / whatever that there's exactly one matching entity and special-case this, better to handle multiple entities for free.

File: src/screens/splash.rs
Comment (will-hart) [reply src/screens/splash.rs]: Yep, happy to change but this is exactly the kind of thing `VecDeque` is in `std::collections` for.

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:189]:
```suggestion
    // Try to get the next splash image.
    let Some(next_splash_image) = splash_images.0.pop_front() else {
        // There are no more splash images, so continue to the loading screen.
```

**Review by benfrankel — CHANGES_REQUESTED** [review]

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:69]: This should keep the original `load_with_settings` and comments from before the PR.

**Review by benfrankel — CHANGES_REQUESTED** [review]

File: src/screens/splash.rs
Comment (benfrankel) [inline src/screens/splash.rs:205]: Change `TimerMode::Once` -> `TimerMode::Repeating`, and then this line isn't needed.

**Review by benfrankel — CHANGES_REQUESTED** [review]

File: src/screens/splash.rs
Comment (janhohenheim) [inline src/screens/splash.rs:106]: Don't think this check is necessary. Doesn't the system with the timer do this for us anyways?

File: src/screens/splash.rs
Comment (janhohenheim) [inline src/screens/splash.rs:57]: Add a doc comment saying why we need this and what it represents.

**Review by janhohenheim — CHANGES_REQUESTED** [review]
Since this veers into the territory of "nice to have, but not necessary for most users", I want @alice-i-cecile's opinion on whether we should add this. I'm personally ambivalent.

In any case, consider upstreaming the code as an official example. I think it would be really helpful there :D

File: src/screens/splash.rs
Comment (alice-i-cecile) [reply src/screens/splash.rs]: Keep it as vec-deque but add a comment about why it's used IMO.

**Comment by alice-i-cecile** [comment]
I like both of these, but strongly seconding the recommendation to upstream the splash screen code as an official example in the `games` folder!

File: src/screens/splash.rs
Comment (will-hart) [reply src/screens/splash.rs]: This has been removed, but this prevents a pause for the timer's duration with a blank screen if no images are provided.

**Comment by will-hart** [comment]
OK, I've removed the multi-splash screen code, resolved any discussions relating that and updated the PR description.

**Review by benfrankel — APPROVED** [review]

File: src/screens/splash.rs
Comment (janhohenheim) [inline src/screens/splash.rs:44]: I personally think that this reads better when written as two separate `run_if` because of the line break, but I'm not blocking on that.

**Review by janhohenheim — APPROVED** [review]

File: src/screens/splash.rs
Comment (benfrankel) [reply src/screens/splash.rs:44]: IMO that's confusing because you have to know whether that yields an `and` or an `or` behavior.

**Pull request merged**
